src/storage/src/source/generator/ldbc.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::BTreeMap;
+use std::io::Read;
+use std::path::PathBuf;
+
+use async_compression::tokio::bufread::ZstdDecoder;
+use async_trait::async_trait;
+use bytes::{Buf, Bytes};
+use csv::{ReaderBuilder, StringRecord};
+use flate2::read::GzDecoder;
+use mz_ore::now::NowFn;
+use mz_repr::{Datum, RelationDesc, Row, RowArena};
+use mz_storage_types::sources::{Generator, MzOffset, LDBC_DESCS, LDBC_PERSON_KNOWS_PERSON_OUTPUT};
+use timely::dataflow::operators::to_stream::Event;
+use tokio_stream::StreamExt;
+use tokio_util::io::StreamReader;
+use tracing::error;
+
+#[derive(Clone, Debug)]
+pub struct Ldbc {
+    pub urls: Vec<String>,
+}
+
+#[async_trait]
+impl Generator for Ldbc {
+    async fn by_seed(
+        &self,
+        _: NowFn,
+        _seed: Option<u64>,
+        _resume_offset: MzOffset,
+    ) -> Box<(dyn Iterator<Item = (usize, Event<Option<MzOffset>, (Row, i64)>)>)> {
+        // TODO: Stream this into the tar reader instead of accumulating all the bytes first.
+        let bytes = match fetch_urls_compressed(&self.urls).await {
```

Comment:
How do we know that these `bytes` will be definite? Can we add the expected hash in the source so that we only release the data in the dataflow if the file we downloaded had the expected hash?

Reply:
If we make all rows the same offset (as discussed above), I think it doesn't really matter right?